File: tests/fixtures/realistic/go_multihop_sanitized/store.go
// Sink helper (fileC) for the go_multihop_sanitized fixture.
//
// Same sink as the positive fixture. The chain is broken upstream (in
// loadFile via filepath.Clean), so no taint finding fires on a directory scan.

package go_multihop

import "os"

func readData(name string) []byte {
	data, _ := os.ReadFile(name)
	return data
}

File: tests/fixtures/realistic/js_multihop_sanitized/store.js
// Sink helper (fileC) for the js_multihop_sanitized fixture.
//
// Same sink as the positive fixture. The chain is broken upstream (in
// service.handle), so the only finding here is the single-file regex heuristic.

const db = { query(_q) { return []; } };

function runQuery(term) {
    return db.query("SELECT * FROM users WHERE name = '" + term + "'");
}

module.exports = { runQuery };
